File: EKI_Invers/src/game/view/Board.ts
class Board {
    private pieces: Array<Piece>;

    constructor() {
        this.pieces = this.initializePieces();
    };

    public getValue(row: number, col: number ): Piece {
        //TODO: Implement
        return this.pieces[0];
    }

    public setValue( row: number, col: number, piece: Piece ) {
        //TODO: Calculate Index Correct
        this.pieces[0] = piece
    }

    public pushLine(row: number, col: number, direction: Direction) {

    }

    public flipField(row: number, col: number) {
        let value = this.getValue(row, col);
        if(value === Piece.RED) {
            this.setValue(row, col, Piece.RED_DOT);
        } else if(value === Piece.YELLOW) {
            this.setValue(row, col, Piece.YELLOW_DOT);
        }
    }

    public getValueForDisplay(row: number, col: number): Piece {
        // TODO: Implement this
        const pieces: Array<Piece> = []
        return pieces[0];
    }

    private initializePieces(): Array<Piece> {
        const pieces: Array<Piece> = [];
        for (let i = 0; i < 100; i++) {
            pieces.push( Piece.EMPTY );
        }
        return pieces;
    };
};

File: invers_backend/src/main/kotlin/de/fherfurt/view/Board.kt
package de.fherfurt.view

import de.fherfurt.core.Direction
import de.fherfurt.core.Piece
import de.fherfurt.model.Player
import java.util.*

class Board {
    val pieces: Vector<Piece>

    init {
        this.pieces = initializePieces()
    }

    fun pushLine(row: Int, col: Int, direction: Direction) {

    }

    fun flipField(row: Int, col: Int) {

    }

    fun getPlayerScore(player: Player) : Int {
        return pieces.count { piece ->
            piece == player.piece || piece == player.dottedPiece
        }
    }

    private fun initializePieces(): Vector<Piece> {
        val pieces = Vector<Piece>()
        for(i in 0..99 ) {
            // Border Above & Below
            if(i < 20 || i >= 80) {
                pieces.add(Piece.BORDER)
            }
            else {
                val col = i % 10
                // Border Left and Right
                if(col == 0 ||col == 1 || col == 8 || col == 9) {
                    pieces.add( Piece.BORDER )
                }
                else { // Inner Empty
                    pieces.add(Piece.EMPTY)
                }
            }
        }
        return pieces
    }
}